### ISSUE TITLE
Feature Request: Disable lambda_unhandled_exception_warning_message

### ISSUE THREAD

**Issue by packplusplus** [aws/aws-lambda-python-runtime-interface-client#139]
The lambda_unhandled_exception_warning_message (`LAMBDA_WARNING: Unhandled exception...`) is ~ 633 chars and adds little value to our debugging. I understand the desire to warn people the runtime may have changed underfoot, and that's why their code is raising exceptions, but I'd like a way to disable it.

A pr from me would probably be something like

```
_disable_unhandled_exception_warning = bool(os.environ.get("AWS_LAMBDA_EXCEPTION_WARNING_DISABLE"))
# SNIP ...
if _disable_unhandled_exception_warning:
        from .lambda_literals import lambda_unhandled_exception_warning_message

        log_sink.log(lambda_unhandled_exception_warning_message, _WARNING_FRAME_TYPE)

log_error(error_result, log_sink)
lambda_runtime_client.post_invocation_error(
    invoke_id, to_json(error_result), to_json(xray_fault)
)
```
near https://github.com/aws/aws-lambda-python-runtime-interface-client/blob/056aa365b6b986b0f413d031f054598064df603e/awslambdaric/bootstrap.py#L217

Perhaps you have a different idea? Or maybe this is a non-starter. Didn't want to clutter PRs before asking.

**Comment by tlinhart** [comment]
Another reason one might want to disable this message is that CloudWatch merges this warning with the actual error and handles it as a single log event. When using ALC with JSON log format, this makes it hard to parse the log event as JSON (in my case it's streamed to Grafana Loki).

![image](https://github.com/user-attachments/assets/5ad8fa7c-cd4b-4666-9ac4-68fdbe8caa08)

**Comment by abillingsley** [comment]
like @tlinhart we are facing this issue with our Cloudwatch logs which are forwarded to DataDog for aggregation.  Not only is the structured logging lost but the ERROR log is reported as an INFO log message making monitoring of "ERROR" logs more difficult

<img width="865" alt="Screenshot 2024-10-09 at 8 26 38 PM" src="https://github.com/user-attachments/assets/4a90960b-916c-4ee9-87a5-84ff7f25f699">
